### PR TITLE
Add epithet/emblems infrastructure

### LINKED ISSUE #206: Add support for epithets
It is not currently possible to interact with epithets (localized name) / emblems (internal name) on the server, due to the functionality being missing.

Tasks required to add support:

- [ ] Add /emblem/get_list
- [ ] Add /emblem/set
- [ ] Add save import support for emblems

This will require creating a new database table to store emblem information.

To follow once other infrastructure is in place:

- [ ] Provide emblems as rewards for completing endeavours or quests

### PR DESCRIPTION
**Pull request by LukeFZ**
closes #206 
apparently they aren't in load/index so no import

~~TODO: savefile update to grant default epithet?~~ added

### CODE REVIEW

File: DragaliaAPI/AutoMapper/Profiles/EmblemReverseMapProfile.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public EmblemReverseMapProfile()
+    {
+        this.AddGlobalIgnore("DeviceAccount");
+        this.AddGlobalIgnore("Owner");
+
+        this.CreateMap<EmblemList, DbEmblem>()
+            .ForMember(x => x.EmblemId, o => o.MapFrom(x => x.emblem_id))
+            .ForMember(x => x.GetTime, o => o.MapFrom(src => src.gettime))
+            .ForMember(x => x.IsNew, o => o.MapFrom(src => src.is_new));
```
Comment (SapiensAnatis) [inline DragaliaAPI/AutoMapper/Profiles/EmblemReverseMapProfile.cs:17]: for emblemId and isNew the naming conventions should be able to figure it out -- or are you just preferring to be explicit

File: DragaliaAPI/Features/Emblem/EmblemController.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    IUserDataRepository userDataRepository,
+    IUpdateDataService updateDataService
+) : DragaliaControllerBase
+{
+    [HttpPost("get_list")]
+    public async Task<DragaliaResult> GetList()
+    {
+        EmblemGetListData resp = new();
+
+        resp.emblem_list = (await emblemRepository.GetEmblemsAsync()).Select(
```
Comment (SapiensAnatis) [inline DragaliaAPI/Features/Emblem/EmblemController.cs:23]: Do we need the mapper profile if we aren't using mapper to go to EmblemList

Comment (LukeFZ) [reply DragaliaAPI/Features/Emblem/EmblemController.cs:23]: I added the mapper profile before realizing the emblem list is in neither load/index nor the update data list, will remove it

File: DragaliaAPI/AutoMapper/Profiles/EmblemReverseMapProfile.cs (hunk 1 above)
Comment (LukeFZ) [reply DragaliaAPI/AutoMapper/Profiles/EmblemReverseMapProfile.cs:17]: I'll remove it

File: DragaliaAPI/Features/SavefileUpdate/V10Update.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace DragaliaAPI.Features.SavefileUpdate;
+
+public class V10Update(IEmblemRepository emblemRepository) : ISavefileUpdate
+{
+    public int SavefileVersion => 10;
+
+    public async Task Apply()
+    {
+        if (!await emblemRepository.HasEmblem(Emblems.DragonbloodPrince))
```
Comment (SapiensAnatis) [inline DragaliaAPI/Features/SavefileUpdate/V10Update.cs:12]: Could we also grant the user's currently equipped emblem?

Comment (LukeFZ) [reply DragaliaAPI/Features/SavefileUpdate/V10Update.cs:15]: added

**Review by SapiensAnatis — APPROVED** [review]

**Review by SapiensAnatis — APPROVED** [review]

**Pull request merged**